workshops/models.py

```diff
@@ -227,9 +228,14 @@ def ongoing_events(self):
         return queryset
 
     def unpublished_events(self):
-        '''Return a queryset for events that are not yet published.'''
+        '''Return events without URLs that are upcoming or have unknown starts.
```

Comment:
To avoid catching folks by surprise, I'd suggest adding `help_text` to `Event.url` explaining that it shouldn't be set until an event is ready to be published.  In my limited experience, there's often a setup period while a partially-completed website is available online, but before you'd actually want to point anyone at it.